wire-library/wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WirePlugin.kt

```diff
@@ -159,17 +160,22 @@ class WirePlugin : Plugin<Project> {
       // Even though we add the Wire output directories into the corresponding sourceSets, the
       // compilation tasks won't know about them so we fix that here.
       if (hasJavaOutput) {
-        project.tasks.matching { it.name == "compileJava" }.configureEach {
-          (it as JavaCompile).source(generatedSourcesDirectories)
-        }
+        project.tasks
+          .withType(JavaCompile::class.java)
```

Comment:
Opportunistically added these to save the cast below. You may even be able to nix the name matching entirely but I wasn't sure how precise those were intended to be